### PR TITLE
added command tests

### PR DESCRIPTION
**Pull request by randrii**

### CODE REVIEW

File: src/test/java/com/rybka/command/ConvertCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @Mock
+    private Scanner scanner;
+    @Mock
+    private ExchangeService exchangeService;
+    @Mock
+    private CurrencyHistoryRepository currencyHistoryRepository;
+    @InjectMocks
+    private ConvertCommand command;
+
+    @Test
```
Comment (jarc0der) [inline src/test/java/com/rybka/command/ConvertCommandTest.java:31]: What about test to verify correct exception handling?

File: src/test/java/com/rybka/command/ExportCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        // given
+        var exportType = "csvs";
+        var command = new ExportCommand(exportProperty, exportConfigMap, currencyHistoryRepository);
+
+        when(currencyHistoryRepository.findAll()).thenReturn(new ArrayList<>());
+        when(exportProperty.getType()).thenReturn(exportType);
+
+        // when
+        command.execute();
```
Comment (jarc0der) [inline src/test/java/com/rybka/command/ExportCommandTest.java:61]: exportType `csvs` is invalid. Looks like the exception should be thrown in the `execute()` method

File: src/test/java/com/rybka/command/ExportCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    public void testOnExportException() {
+
+        // given
+        var incorrectExportType = "xml";
+        var command = new ExportCommand(exportProperty, exportConfigMap, currencyHistoryRepository);
+
+        when(currencyHistoryRepository.findAll()).thenReturn(new ArrayList<>());
```
Comment (jarc0der) [inline src/test/java/com/rybka/command/ExportCommandTest.java]: `List.of()` should works here.

File: src/test/java/com/rybka/command/ConvertCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        var currencyAmount = 5d;
+        var currencyRate = 0.8877;
+        var currencyData = new CurrencyData(baseCurrency, targetCurrency, currencyRate);
+        var currencyHistory = new CurrencyHistory();
+
+        when(scanner.next()).thenReturn(baseCurrency).thenReturn(targetCurrency);
+        when(scanner.nextDouble()).thenReturn(currencyAmount);
+        when(exchangeService.loadCurrencyOf(baseCurrency, targetCurrency)).thenReturn(currencyData);
+        when(exchangeService.calculateTotal(currencyRate, currencyAmount)).thenReturn(currencyRate * currencyAmount);
+        when(currencyHistoryRepository.save(Mockito.any(CurrencyHistory.class))).thenReturn(currencyHistory);
```
Comment (jarc0der) [inline src/test/java/com/rybka/command/ConvertCommandTest.java:46]: Seems it's hard to check the parameter that has been thrown to the currencyHistoryRepository.save(...) method. 
But we can do that. Take a look and `ArgumetCaptor` in the `Mockito` library, it allows to catch parameters and add additional verifications.

File: src/test/java/com/rybka/command/ExportCommandTest.java (hunk 3 above)
Comment (randrii) [reply src/test/java/com/rybka/command/ExportCommandTest.java]: fixed

File: src/test/java/com/rybka/command/ExportCommandTest.java (hunk 2 above)
Comment (randrii) [reply src/test/java/com/rybka/command/ExportCommandTest.java:61]: fixed

File: src/test/java/com/rybka/command/ConvertCommandTest.java (hunk 4 above)
Comment (randrii) [reply src/test/java/com/rybka/command/ConvertCommandTest.java:46]: fixed

File: src/test/java/com/rybka/command/ConvertCommandTest.java (hunk 1 above)
Comment (randrii) [reply src/test/java/com/rybka/command/ConvertCommandTest.java:31]: fixed

**Pull request merged**